File: src/basic_training/Day11_3.java
package basic_training;

import java.util.ArrayList;
import java.util.List;

public class Day11_3 {
	public String solution(String my_string, int[] indices) {
		StringBuilder sb = new StringBuilder();
		List<Integer> intArr = new ArrayList<>();
		for(int idx : indices) {
			intArr.add(idx);
		}
		for(int i=0; i<my_string.length(); i++){
			if(!intArr.contains(i)) { 
				sb.append(my_string.charAt(i));
			}
		}	
		String answer = sb.toString();
		return answer;
	}
}

// 🍒 자바 List배열 contains() 메서드
// List 클래스에서 제공하는 contains() 메서드. 배열에 특정 값이 존재하는지 확인하는 법

// 🍉 참고할 만한 번외 문제풀이
//  StringBuilder sb = new StringBuilder(my_string);
// 	for (int i : indices) {
// 			sb.setCharAt(i, ' ');
// 	}
// 	return sb.toString().replace(" ", "");

// 🍉 참고할 만한 번외 문제풀이
// String[] str = my_string.split("");
// for (int i=0;i<indices.length;i++) str[indices[i]] = "";
// return String.join("",str);

// String.join() 예시
// String[] arr = {"", "", "b", "na", "na"};
// String result = String.join("", arr); // arr 배열의 빈문자열과 arr 배열요소를 문자열로 합치는것
// System.out.println(result); // 출력: "bnana"

File: bin/basic_training/Day11_3.js
function solution(my_string, indices) {
	return [...my_string].filter((value, index) => !indices.includes(index)).join('');
}

// 🍉 문제풀이 _ 배열 forEach / join()
	// const strArr = [...my_string];
	// indices.forEach(value => strArr[value] = '');
	// return strArr.join('');

// 🍉 번외 문제풀이 _ 배열 map / includes / join()
// const result = [...my_string].map((value, index) => {
// 	if(indices.includes(index)) {return ''}
// 	return value;
// }).join('');
// return result;

// 🍉 번외 문제풀이 _ 배열 for of 문 / join()
// const strArr = my_string.split('');
// for(let idx of indices) {
// 	strArr[idx] = '';
// }
// return strArr.join('');